src/main/java/net/snowflake/client/core/HttpClientSettingsKey.java

```diff
@@ -15,35 +15,34 @@
 public class HttpClientSettingsKey implements Serializable {
 
   private OCSPMode ocspMode;
-  private boolean useProxy;
+  private boolean useProxy = false;
   private String proxyHost = "";
   private int proxyPort = 0;
   private String nonProxyHosts = "";
   private String proxyUser = "";
   private String proxyPassword = "";
   private String proxyProtocol = "http";
+  private String userAgentSuffix = "";
 
   public HttpClientSettingsKey(
```

Comment:
Can we create a new overloaded constructor so no need to change in the caller and it's public, no need to worry about breaking change.

Reply:
It looks like we already have quite a few number of potential settings that we are changing via constructor overload. In this case I have to write 2 new overloads for usage proxy and the regular case.
I think that builder pattern is a better approach to manage the parameters and enforce their compatibility unless we want  backward compatibility due to widespread external dependencies.

Reply:
Sent a new iteration where I'm keeping the public constructor.